cdap-ui/app/cdap/components/AbstractWidget/InputFieldDropdown/index.tsx

```diff
@@ -18,16 +18,24 @@ import Select from 'components/AbstractWidget/FormInputs/Select';
 import { IWidgetProps, IStageSchema } from 'components/AbstractWidget';
 import { objectQuery } from 'services/helpers';
 import { WIDGET_PROPTYPES } from 'components/AbstractWidget/constants';
+import MultiSelect from '../FormInputs/MultiSelect';
 
 interface IField {
   name: string;
-  type: string;
+  type: string[];
 }
 
-interface IInputFieldProps extends IWidgetProps<null> {}
+const delimiter: string = ',';
+
+interface IInputFieldWidgetProps {
+  multiselect?: boolean;
+  allowedTypes?: string[];
```

Comment:
I am.not sure if I follow what is allowedtypes here. This widget takes fields from input schema and shows in a drop-down. What would allowed types mean in that context . May be I am missing something.

Reply:
Added a comment to describe this change better